index.js

```diff
@@ -160,10 +165,20 @@ Filter.prototype.build = function() {
             return fs.unlinkSync(outputPath);
           } case 'change': {
             instrumentation.change++;
-            return plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            var changeOperation = plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            if (plugin.async) {
```

Comment:
One downside would be memory usage in plugins that use this filter. For the project I'm working on, there is a babel transpilation step that is 3000+ operations, which are all started and queued up to finish async. If those operations each reserve a lot of memory, that can overflow the JS heap - I caused that to happen a couple times.

I think that is something that should be handled in the plugins that use this filter, by using memory more intelligently, or perhaps throttling. I don't think that should be handled here, to keep this more general-purpose.